### PR TITLE
added assignment4

### PR DESCRIPTION
**Pull request by shivani-tr**

### CODE REVIEW

File: assignment2/src/components/core/ProductTable.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (gauravvolopay) [inline assignment2/src/components/core/ProductTable.jsx:1]: Use the same Table core component which u have use for losers and gainers

File: assignment2/src/components/core/TableButtons.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
```
Comment (gauravvolopay) [inline assignment2/src/components/core/TableButtons.jsx:2]: Remove it from the core folder, it is not a core component

File: assignment2/src/components/InfiniteScroll.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+
+import React, { useEffect, useRef, useCallback } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { fetchAllProducts, incrementPage } from '../store/reducers/productSlice.js';
+import ProductTable from "./core/ProductTable";
+
+const InfiniteScroll = () => {
+    const dispatch = useDispatch();
+    const products = useSelector((state) => state.product.products);
```
Comment (gauravvolopay) [inline assignment2/src/components/InfiniteScroll.jsx:9]: Create a selectors file then use those selectors here to get the values here

**Review by gauravvolopay — CHANGES_REQUESTED** [review]

File: assignment2/src/components/core/ProductTable.jsx (hunk 1 above)
Comment (shivani-tr) [reply assignment2/src/components/core/ProductTable.jsx]: Updated Table.jsx to be reusable

File: assignment2/src/components/core/TableButtons.jsx (hunk 2 above)
Comment (shivani-tr) [reply assignment2/src/components/core/TableButtons.jsx]: Moved TableButtons.jsx inside products folder inside components

File: assignment2/src/components/InfiniteScroll.jsx (hunk 3 above)
Comment (shivani-tr) [reply assignment2/src/components/InfiniteScroll.jsx]: a seperate ProductsSelectors.js created

**Review by gauravvolopay — APPROVED** [review]

**Pull request merged**